integration-tests/test_register.py

```diff
@@ -0,0 +1,136 @@
+
+Special usecases for registering (with proxy, activation keys, ...) are presented
+in its own files.
+
+It is important to run tests as root. Since RegisterServer is a system dbus service.
+And it provides a unix socket connection.
+"""
+
+
+def test_register(any_candlepin, subman, test_config):
+    """
+    https://www.candlepinproject.org/docs/subscription-manager/dbus_objects.html#methods-6
+    """
+    assert not subman.is_registered
+    proxy = RHSM.get_proxy(RHSM_REGISTER_SERVER)
+    with RHSMPrivateBus(proxy) as private_bus:
+        private_proxy = private_bus.get_proxy(RHSM.service_name, RHSM_REGISTER.object_path)
+        locale = os.environ.get("LANG", "")
+        response = private_proxy.Register(
+            "",
+            test_config.get("candlepin", "username"),
+            test_config.get("candlepin", "password"),
+            {},
+            {},
+            locale,
+        )
+        response_data = json.loads(response)
+        assert "idCert" in response_data, "A response contains of consumer certificate"
+        assert frozenset(["key", "cert", "updated", "created", "id", "serial"]).issubset(
+            frozenset(response_data["idCert"].keys())
+        )
+
+    assert subman.is_registered
+
+
+def test_register_with_org(external_candlepin, subman, test_config):
+    """
+    https://www.candlepinproject.org/docs/subscription-manager/dbus_objects.html#methods-6
+    """
+    assert not subman.is_registered
+    proxy = RHSM.get_proxy(RHSM_REGISTER_SERVER)
+    with RHSMPrivateBus(proxy) as private_bus:
+        private_proxy = private_bus.get_proxy(RHSM.service_name, RHSM_REGISTER.object_path)
+        locale = os.environ.get("LANG", "")
+        private_proxy.Register(
+            test_config.get("candlepin", "org"),
+            test_config.get("candlepin", "username"),
+            test_config.get("candlepin", "password"),
+            {},
+            {},
+            locale,
+        )
+    assert subman.is_registered
+
+
+@pytest.mark.parametrize("enable_content", ["true", "false"])
+def test_register_with_enable_content(external_candlepin, subman, test_config, enable_content):
+    """
+    https://www.candlepinproject.org/docs/subscription-manager/dbus_objects.html#methods-6
+    """
+    assert not subman.is_registered
+    proxy = RHSM.get_proxy(RHSM_REGISTER_SERVER)
+    with RHSMPrivateBus(proxy) as private_bus:
+        private_proxy = private_bus.get_proxy(RHSM.service_name, RHSM_REGISTER.object_path)
+        locale = os.environ.get("LANG", "")
+        private_proxy.Register(
+            test_config.get("candlepin", "org"),
+            test_config.get("candlepin", "username"),
+            test_config.get("candlepin", "password"),
+            {"enable_content": get_variant(Str, enable_content)},
+            {},
+            locale,
+        )
+    assert subman.is_registered
+
+
+@pytest.mark.parametrize("wrong_case", ["wrong username", "wrong password", "wrong organization"])
```

Comment:
I can see what you try to test, but following code is little bit confusing. Try to change it little bit to be more descriptive. Please use something like this:

```python
@pytest.mark.parametrize("wrong_case", [("wrong username", None, None), (None, "wrong password", None), (None, None, "wrong organization")])
def test_wrong_something(wrong_case):
    print(f"username: {wrong_case[0]}")
    print(f"password: {wrong_case[1]}")
    print(f"organization: {wrong_case[2]}")
```

This approach will allow us to use more wrong combinations in the future.

Reply:
I updated it to use a standar way of parameters. It looks odd but no problem with that:
PASSED test_register.py::test_register_with_wrong_values[credentials0]
PASSED test_register.py::test_register_with_wrong_values[credentials1]
PASSED test_register.py::test_register_with_wrong_values[credentials2]

btw: is is easy to combine arguments when uses a plain string aka: 'wrong username, wrong password", "wrong password, wrong org", ...

pytest output is more readable then:
PASSED test_register.py::test_register_with_wrong_values["wrong username, wrong password"]
PASSED test_register.py::test_register_with_wrong_values["wrong username"]
PASSED test_register.py::test_register_with_wrong_values["wrong username, wrong organization"]

ok, I've rewritten the code to use a common way to parametrize tests.